subprojects/platform-base/src/main/java/org/gradle/language/base/internal/tasks/SimpleStaleClassCleaner.java

```diff
@@ -31,7 +31,7 @@ public SimpleStaleClassCleaner(TaskOutputsInternal taskOutputs) {
     public void execute() {
         String prefix = getDestinationDir().getAbsolutePath() + File.separator;
         for (File f : taskOutputs.getPreviousOutputFiles()) {
-            if (f.getAbsolutePath().startsWith(prefix)) {
+            if (f.getAbsolutePath().startsWith(prefix) && f.isFile()) {
```

Comment:
Deleting directories is more difficult, since we would need to check for overlapping outputs, again, in order not to delete files from another task. I would just not invest more time into the `SimpleStaleClassCleaner`. WDYT?